src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -261,6 +261,37 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
+    ProjectReference Targets: Include transitive project references before 
+                              ResolveProjectReferences is called
+    - _ComputeTransitiveProjectReferences
+    - IncludeTransitiveProjectReferences
+    ============================================================
+    -->
+
+  <Target Name="_ComputeTransitiveProjectReferences"
+          DependsOnTargets="RunResolvePackageDependencies"
+          Returns="_TransitiveProjectReferences">
+    <ItemGroup>
+      <ProjectDefs Include="@(PackageDefinitions->WithMetadataValue('Type', 'project'))" />
+
+      <_TransitiveProjectReferences Include="%(ProjectDefs.Path)">
+        <ResolvedPath>%(ProjectDefs.ResolvedPath)</ResolvedPath>
+      </_TransitiveProjectReferences>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="IncludeTransitiveProjectReferences"
+          Condition="'$(ResolvePackageDependenciesForBuild)' == 'true' and '$(EnableResolvePackageDependencies)' == 'true' and Exists('$(ProjectAssetsFile)')"
+          DependsOnTargets="_ComputeTransitiveProjectReferences" 
+          BeforeTargets="AssignProjectConfiguration">
```

Comment:
It appears the two flag properties are redundant, and may have been added separately. I've removed `EnableResolvePackageDependencies` and also simplified the ordering as you suggested to avoid repeating the condition.